docs/DeveloperGuide.md

```diff
@@ -1,22 +1,241 @@
+How `UI` Works:
+1. Whenever the program needs to interact with the user, it does so through the `UI` class, which serves as a **facade** 
+    between the logical backend components and the user interface elements.
+
+2. The `UI` class is responsible for displaying messages, receiving user input, and printing data formatted as tables or 
+text responses.
+
+3. Upon receiving a message or command request, the UI class formats the message, incorporating relevant content 
+(e.g., student details or allocation results) before displaying it to the user. This approach maintains separation 
+between backend logic and the presentation layer.
+
+4. For each user action, such as displaying a list of students or allocating slots, the `UI` class uses helper methods 
+(e.g., `printStudentList`, `generateReport`) to format and render the output. The methods ensure the responses are 
+consistent and user-friendly.
+
+The `UI` class methods typically return `void`, but will print responses directly to the console or handle user input, 
+streamlining interactions and allowing the backend to focus solely on data processing.
+
+#### 2. `Messages` Enum
+   The `Messages` enum centralizes common UI messages. For example, `Messages.ERROR` is passed to the UI for 
+   display for default errors. This keeps responses uniform and allows for changes to user-facing text without modifying 
+   backend logic. List of all `Messages`:
+
+`WELCOME`: Greeting message displayed at startup.
+
+`EXIT`: Farewell message when exiting.
+
+`ALLOCATE_COMPLETE`: Shown upon completion of the allocation process.
+
+`HELP`: Multi-line help message listing all available commands and their descriptions.
+
+`ERROR`: General error message for unexpected issues.
+
+`REVERT_COMPLETE`: Message displayed after a successful revert operation.
+
+Each message can be accessed and printed via `Messages.<MESSAGE_NAME>` in the `UI` class or any other class that 
+references it.
+
+#### Customizing and Extending the UI
+   Adding New Messages:
+1.  Open the `Messages` enum.
+2.  Add a new constant with the message text. Example:
+```java
+NEW_MESSAGE("Your custom message text here");
+```
+3. Reference the new message in the `UI` class or any other relevant part of the application using `Messages.NEW_MESSAGE`
+
+
+
+### Parser
+
+The `Parser` class is a crucial component instantiated as soon as FindOurSEP is initialized. Its responsibility is to process the user’s input into commands and invoking the correct command object for the rest of the program.
+
+Some of its core features include:
+- Breaking down user input and extracting the relevant command and data for further processing.
+- Provide robust error handling when unknown command is received.
+- Validate data parsed in from external file (.CSV, .JSON, .TXT) sources. (Further details in the `Storage` class)
+
+Here is a class diagram highlighting the fundamental structure of the `Parser` class.
+
+![ParserClass](UML_Diagrams/ParserClass.drawio.svg)
```

Comment:
![image](https://github.com/user-attachments/assets/20b5fef5-d153-4741-bf3a-2880dca634a4)

You could consider segmenting out the {abstract} Command class to improve its visibility, it currently looks like another command class.